vmm/src/config.rs

```diff
@@ -2660,6 +2781,23 @@ impl VmConfig {
             }
             disks = Some(disk_config_list);
         }
+        #[cfg(feature = "fw_cfg")]
+        let fw_cfg_config = if vm_params.fw_cfg_config.is_some() {
+            let kernel = vm_params.kernel.is_some();
+            let cmdline = vm_params.cmdline.is_some();
+            let initramfs = vm_params.initramfs.is_some();
+            let fw_cfg_config = FwCfgConfig::parse(vm_params.fw_cfg_config.unwrap())?;
+            if fw_cfg_config.kernel && !kernel {
+                return Err(Error::ParseFwCfgConfigKernel);
+            } else if fw_cfg_config.cmdline && !cmdline {
+                return Err(Error::ParseFwCfgConfigCmdline);
+            } else if fw_cfg_config.initramfs && !initramfs {
+                return Err(Error::ParseFwCfgConfigInitramfs);
+            }
```

Comment:
Let's separate validate from parse - add a `FWCfgCongfig::validate()` for this and call it from `VmConfig::validate()`.